cli/loader/src/lib.rs

```diff
@@ -521,11 +521,20 @@ impl Loader {
             .and_then(|n| n.to_str())
             .and_then(|file_name| self.language_configuration_ids_by_file_type.get(file_name))
             .or_else(|| {
-                path.extension()
-                    .and_then(|extension| extension.to_str())
-                    .and_then(|extension| {
-                        self.language_configuration_ids_by_file_type.get(extension)
-                    })
+                let mut path = path.to_owned();
+                let mut extensions = Vec::new();
+                while let Some(extension) = path.extension() {
+                    extensions.push(extension.to_str()?.to_string());
+                    path = PathBuf::from(path.file_stem()?.to_os_string());
+                }
+                let full_extension = extensions
```

Comment:
This felt a bit awkward, but using `VecDeque` with `push_front` doesn't work since `.join` isn't implemented on that container. I tried two approaches but left them in separate commits.